onnxruntime/core/framework/kernel_type_str_resolver.h

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <string>
+#include <string_view>
+#include <utility>
+
+#include "gsl/gsl"
+
+#if !defined(ORT_MINIMAL_BUILD)
+#include "onnx/defs/schema.h"
+#endif  // !defined(ORT_MINIMAL_BUILD)
+
+#include "core/common/inlined_containers.h"
+#include "core/common/status.h"
+#include "core/graph/op_identifier.h"
+#include "core/graph/graph.h"
+#include "core/platform/ort_mutex.h"
+
+namespace flatbuffers {
+class FlatBufferBuilder;
+template <typename T>
+struct Offset;
+}  // namespace flatbuffers
+
+namespace onnxruntime {
+
+namespace fbs {
+struct KernelTypeStrResolver;
+}  // namespace fbs
+
+using ArgTypeAndIndex = std::pair<ArgType, size_t>;
+using KernelTypeStrToArgsMap = InlinedHashMap<std::string, InlinedVector<ArgTypeAndIndex>>;
```

Comment:
>InlinedHashMap<std::string, InlinedVector<ArgTypeAndIndex>>; [](http://example.com/codeflow?start=31&length=60)

Be careful using these types cross DLL, as lookups are not likely to work since separate DLLs would have its own version of Abseil.

Reply:
thanks for the tip. `InlinedHashMap` isn't directly passed across the DLL boundary here. and now it's going through another IKernelLookup interface. let me know if you see any problems there.